### PR TITLE
Allow the user to specify multiple paths to codebase

### PR DESCRIPTION
**Pull request by ketkarameya**
Allow user to specify multiple paths to codebase. 
* Refactored the core "file-fetching" logic 
* Added tests

### CODE REVIEW

File: src/models/rule_store.rs [hunk 1]
```diff
@@ -143,19 +143,58 @@ impl RuleStore {
...
-    if _path_to_codebase.is_file() {
-      return HashMap::from_iter([(
-        _path_to_codebase.clone(),
-        read_file(&_path_to_codebase).unwrap(),
-      )]);
+    let mut candidate_files: HashMap<PathBuf, String> = HashMap::new();
+
+    for p2codebase in paths_to_codebase {
+      if !Path::new(p2codebase).exists() {
+        panic!("Path to codebase does not exist: {}", p2codebase);
```
Comment (lazaroclapp) [inline src/models/rule_store.rs:152]: Are we testing for this error condition anywhere in our test suite? Seems new and not in the one newly added test case.

**Review by lazaroclapp — COMMENTED** [review]
Some nits, but mostly looks good to me :)

File: src/models/rule_store.rs (hunk 1 above)
Comment (ketkarameya) [reply src/models/rule_store.rs:152]: added a test

File: src/models/piranha_arguments.rs [hunk 2]
```diff
@@ -50,9 +50,9 @@ use std::collections::HashMap;
 pub struct PiranhaArguments {
   /// Path to source code folder or file
   #[get = "pub"]
-  #[builder(default = "default_path_to_codebase()")]
-  #[clap(short = 'c', long, required = true)]
-  path_to_codebase: String,
+  #[builder(default = "default_paths_to_codebase()")]
+  #[clap(short = 'c', long, num_args = 0.., required = true)]
```
[the remark this answers is not in the document]
Comment (ketkarameya) [reply src/models/piranha_arguments.rs:54]: User need not specify the `path_to_codebase` with the `code_snippet` option

**Review by ketkarameya — COMMENTED** [review]
addressed

**Review by lazaroclapp — APPROVED** [review]
LGTM!

**Pull request merged**